C++/source/searching/Exponential_search.cpp

```diff
@@ -0,0 +1,66 @@
+// C++ program to find an element x in a
+// sorted array using Exponential search.
+#include <bits/stdc++.h>
+using namespace std;
+ 
+int binarySearch(int arr[], int, int, int);
+ 
+// Returns position of first ocurrence of
+// x in array
+int exponentialSearch(int arr[], int n, int x)
+{
+    // If x is present at firt location itself
+    if (arr[0] == x)
+        return 0;
+ 
+    // Find range for binary search by
+    // repeated doubling
+    int i = 1;
+    while (i < n && arr[i] <= x)
+        i = i*2;
+ 
+    //  Call binary search for the found range.
+    return binarySearch(arr, i/2, min(i, n), x);
+}
+ 
+// A recursive binary search function. It returns
+// location of x in  given array arr[l..r] is
+// present, otherwise -1
+int binarySearch(int arr[], int l, int r, int x)
```

Comment:
Binary search [already exists](https://github.com/faheel/Algos/blob/master/C%2B%2B/source/searching/binary_search.cpp). It would be better if you found a way of reusing it here, rather than reimplementing it.